database.py

```diff
@@ -34,34 +38,44 @@ def create_table_if_needed(cursor):
         cursor.execute('''CREATE TABLE osm_bot_edit_log
                     (area_identifier text, type text, bot_edit_timestamp integer)''')
 
-def existing_tables(cursor):
+
+def existing_tables(cursor: Cursor):
     cursor.execute("SELECT name FROM sqlite_master WHERE type='table';")
     table_listing = cursor.fetchall()
     returned = []
     for entry in table_listing:
         returned.append(entry[0])
     return returned
 
-def get_data_download_timestamp(cursor, internal_region_name):
-    cursor.execute("SELECT download_timestamp FROM osm_data_update_log WHERE area_identifier = :area_identifier ORDER BY download_timestamp DESC LIMIT 1", {"area_identifier": internal_region_name})
+
+def get_data_download_timestamp(cursor: Cursor, internal_region_name: string):
+    cursor.execute(
+        "SELECT download_timestamp FROM osm_data_update_log WHERE area_identifier = :area_identifier ORDER BY download_timestamp DESC LIMIT 1",
+        {"area_identifier": internal_region_name})
     returned = cursor.fetchall()
     if len(returned) == 0:
         return 0
     else:
         return returned[0][0]
 
-def get_bot_edit_timestamp(cursor, internal_region_name, bot_edit_type):
-    cursor.execute("SELECT bot_edit_timestamp FROM osm_bot_edit_log WHERE area_identifier = :area_identifier AND type = :type ORDER BY bot_edit_timestamp DESC LIMIT 1", {"area_identifier": internal_region_name, "type": bot_edit_type, })
+
+def get_bot_edit_timestamp(cursor: Cursor, internal_region_name: string, bot_edit_type: string):
```

Comment:
are enums available as types? `bot_edit_type` is more restrictive than just a string, is it possible to express it?

Reply:
Sure, we should be able to do that, is it correct the strings would be 
- handle_follow_wikipedia_redirect_where_target_matches_wikidata
- handle_wikidata_redirect
- add_wikipedia_tag_from_wikidata_tag
- add_wikidata_tag_from_wikipedia_tag

Reply:
wait, here it would be quite annoying as any new bot edit would require fiddling

Reply:
I agree :)